syntaxes/alex.YAML-tmLanguage

```diff
@@ -0,0 +1,75 @@
+fileTypes:
+      - x
+name: Haskell Alex
+patterns:
+  - include: "#comments"
+  - include: "#blocks"
+  - include: "#syntax"
+  - include: "#strings"
+repository:
+  comments:
+    patterns:
+      - name: comment.line
+        match: \-\-.*$
+      - name: comment.block
+        begin: "{-"
+        end: "-}"
+  blocks:
+    name: meta.embedded.block.haskell
+    begin: "{"
+    beginCaptures:
+      "0": { name: punctuation.block.begin.alex }
+    end: "}"
+    endCaptures:
+      "0": { name: punctuation.block.end.alex }
+    patterns:
+      - include: source.haskell
```

Comment:
I don't think this is quite right, as braces can directly follow a start-code to indicate multiple rules over several lines, and don't indicate Haskell code. For instance, from GHC's `Lexer.x`:

```alex
<bol> {
  \n                                    ;
  ^\# line                              { begin line_prag1 }
  ^\# / { followedByDigit }             { begin line_prag1 }
  ^\# pragma .* \n                      ; -- GCC 3.3 CPP generated, apparently
  ^\# \! .* \n                          ; -- #!, for scripts
  ()                                    { do_bol }
}
```

The outermost braces do not herald Haskell source.

Reply:
Great point, but this one is a bit of a puzzler for me as to solving it... Should I make a production for `<startcode> <open bracket> <nested context> <end bracket>` or something of the like?

Reply:
Personally I would go for a `begin`/`end` pair, where the `begin` rule is given by the regular expression:

```(<)\s*(\w*)(?:\s*(,)\s*(\w*))*\s*(>)\s*(\{)```

That would capture all the start codes, allowing spaces in any position as well as providing scopes for the various bits of punctuation (angled brackets, commas, brace).